### DISCUSSION TITLE
Workflows: Multi-author support

### DISCUSSION THREAD

**Discussion by leogermani** [WordPress/gutenberg#56954] [General]
As part of the [Workflows discussion](https://make.wordpress.org/core/2023/07/04/workflows/), which is part of Phase 3, the possibility of supporting multiple authors per post has been raised.

I’m very much interested in this discussion and wanted to know if there are any previous threads around this, or any consolidated proposal on how this would be supported on core.

I currently work at [Newspack](https://newspack.com) and our clients make sophisticated use of multiple-authors in many different editorial workflows. We mainly use the Co Authors Plus plugin for that. It would be awesome to form a collaboration group to look at the alternatives and find a way forward.

I have a number of question, ideas and proposals on what could be a good approach for us, and wanted to kick off this discussion. I am mostly a back end developer so that's where I'm looking at things from.

## DB strategy

Co Authors plus uses a taxonomy strategy, where each author is also a term. Other plugins store authors as post meta. Even though I think taxonomy is a good strategy for a plugin, I wonder if we are open to consider a new table in our DB schema to store the relationship between posts and users.

This could be a straight forward solution and give us more room to accommodate the requirements that real-time collaboration might raise. For example, a user could have a different role per post?

Such a table could have `post_id`, `user_id`, `order`, `user_role` for example. And we could add some room for plugin customization via a related `_meta` table or a more generic `meta` field.

## Capabilities

How else would this impact capability checks? An author can only edit its own posts. Could they be invited to collaborate on someone else’s post and than gain access to only that post? 

## Collaboration and authorship

Workflows are about collaboration, so we might have many authors working on a post. But a user who helped edit a post is not necessarily an author that should be displayed in the by lines in the front-end of the site. How an editor could control that?

## Backward compatibility with themes

If we update template tags and blocks, how do we keep backward compatibility? Maybe using a `theme_supports => multi-author` to start could be a good idea.

## Guest Authors

It's always raised the need to be able to have an author who is not a WordPress user, but simply a profile in the site. Current plugins usually refer to this as "Guest Authors" and Co Authors plus store them as Custom Post types. (I am a believer that a simple user role without any privileges is a much better solution here, Guest Authors have proven to be the source of may troubles we've had)

Usually these are just bylines in the post and do not add any special access for an external user. But is this a requirement for real-time collaboration? Would we open for non WP users to make edits to one specific post?

## Real-time collaboration

What other requirement real-time collaboration brings that will impact how we store and retrieve authorship in the back-end?

## REST API

What are the impacts on the REST API responses and how we keep backward compatibility?

## Feature plugin

Would a feature plugin be a good way to prototype this change?

And much more... There's a lot to discuss on this topic and I would love to see this moving forward. And I'm willing to collaborate as well. Every now and then e talk about having a better solution and it would be awesome to work towards something that could contribute to the evolution of Core.

**Comment by scottfennell-toptal** [comment]
Co Authors Plus is such a de facto standard here that I think the preference should be to merge it into core.  It would be a mess to have core competing with the way it's generally been done (and done well) for many years.
